src/tess_sip/tess_sip.py

```diff
@@ -92,65 +99,101 @@ def SIP(
+                [
+                    tpf.to_lightcurve(aperture_mask=aper)
+                    for tpf, aper in zip(tpfs_uncorr, apers)
+                ]
+            )
+            .stitch(lambda x: x)
+            .normalize()
         )
-        .stitch(lambda x: x)
-        .normalize()
-    )
-    lc.flux_err.value[~np.isfinite(lc.flux_err.value)] = np.nanmedian(lc.flux_err.value)
+        lc.flux_err.value[~np.isfinite(lc.flux_err.value)] = np.nanmedian(lc.flux_err.value)
 
-    # Run the same routines on the background pixels
-    lc_bkg = (
-        lk.LightCurveCollection(
-            [
-                tpf.to_lightcurve(aperture_mask=bkg_aper)
-                for tpf, bkg_aper in zip(tpfs_uncorr, bkg_apers)
-            ]
+        # Run the same routines on the background pixels
+        lc_bkg = (
+            lk.LightCurveCollection(
+                [
+                    tpf.to_lightcurve(aperture_mask=bkg_aper)
+                    for tpf, bkg_aper in zip(tpfs_uncorr, bkg_apers)
+                ]
+            )
+            .stitch(lambda x: x)
+            .normalize()
         )
-        .stitch(lambda x: x)
-        .normalize()
-    )
-    lc_bkg.flux_err.value[~np.isfinite(lc_bkg.flux_err.value)] = np.nanmedian(
-        lc_bkg.flux_err.value
-    )
+        lc_bkg.flux_err.value[~np.isfinite(lc_bkg.flux_err.value)] = np.nanmedian(
+            lc_bkg.flux_err.value
+        )
+        
+    # Check if tpfs is a lightcurve file rather than a target pixel file
+    elif ((type(tpfs) is lk.lightcurve.TessLightCurve) or 
+            (type(tpfs) is lk.collections.LightCurveCollection)):
+        print("Lightcurve File Input")
+        for lcf in tpfs:
+            lcf.sap_flux.value[~np.isfinite(lcf.sap_flux.value)] = np.nanmedian(lcf.sap_flux.value)
+            lcf.sap_bkg.value[~np.isfinite(lcf.sap_bkg.value)] = np.nanmedian(lcf.sap_bkg.value)
+            lcf.flux = lcf.sap_flux
+            lcf.flux_err = lcf.sap_flux_err
+        tpfs_uncorr = [(lcf + np.nan_to_num(lcf.sap_bkg))[np.isfinite(lcf.sap_bkg)] for lcf in tpfs]
+
+        lc = lk.LightCurveCollection(tpfs_uncorr).stitch(lambda x:x).normalize()
+        lc.flux_err.value[~np.isfinite(lc.flux_err.value)] = np.nanmedian(lc.flux_err.value)
 
+        lc_bkg = lk.LightCurveCollection(tpfs_uncorr).stitch(lambda x:x)
+        lc_bkg.flux = lc_bkg.sap_bkg
+        lc_bkg.flux_err = lc_bkg.sap_bkg_err
+        lc_bkg = lc_bkg.normalize()
+        lc_bkg.flux_err.value[~np.isfinite(lc_bkg.flux_err.value)] = np.nanmedian(lc_bkg.flux_err.value)
+        
     with warnings.catch_warnings():
         warnings.simplefilter("ignore")
-        bkgs = [
-            lk.correctors.DesignMatrix(tpf.flux.value[:, bkg_aper], name="bkg")
-            .pca(npca_components)
-            .append_constant()
-            .to_sparse()
-            for tpf, bkg_aper in zip(tpfs_uncorr, bkg_apers)
-        ]
+        if ((type(tpfs) is lk.collections.TargetPixelFileCollection) or 
```

Comment:
Why is this down here in another `if` statement, I believe you could add this to the end of the previous `if` statement which checks if the input is TPFs.